### PR TITLE
fix(storage-proofs): double expansion parents using permutation inverse

### PR DESCRIPTION
**Pull request by schomatis**
Closes https://github.com/filecoin-project/rust-fil-proofs/issues/827.
Closes https://github.com/filecoin-project/rust-fil-proofs/pull/859.
Closes https://github.com/filecoin-project/rust-fil-proofs/issues/857.

### CODE REVIEW

File: storage-proofs/src/zigzag_drgporep.rs [hunk 1]
```diff
@@ -199,16 +199,14 @@ mod tests {
             // prove_verify_32_3_1(32, 3, 1);
             // prove_verify_32_3_2(32, 3, 2);
 
-           prove_verify_fixed_32_5_1(5, 1);
            prove_verify_fixed_32_5_2(5, 2);
-           prove_verify_fixed_32_5_3(5, 3);
+           prove_verify_fixed_32_5_4(5, 4);
```
Comment (schomatis) [inline storage-proofs/src/zigzag_drgporep.rs]: Changing these just to use an even expansion degree (don't know how the tests work or if this is fine to do).

Comment (dignifiedquire) [reply storage-proofs/src/zigzag_drgporep.rs]: that is fine

**Comment by dignifiedquire** [comment]
@schomatis needs conflict resolution with `master`

Are you planning to write a test or not? (not sure how to interpret your :+1: on my comment)

**Comment by schomatis** [comment]
I'm waiting on confirmation from @porcuquine first (https://github.com/filecoin-project/rust-fil-proofs/issues/827#issuecomment-527146852) if this is the right way to handle the parents increase. (:+1:: test sounds like a good idea, I need to think a bit how to implement it, but after confirmation)

**Comment by porcuquine** [comment]
I commented on #827. I think it probably makes most sense to double the expansion degree rather than require it to be even. If we do this, the value of the expansion degree will not increase, but we will (correctly) generate more actual expansion parents for each node. See #827 for a bit more detail about how `degree` would be calculated in this model.

**Comment by schomatis** [comment]
@dignifiedquire @porcuquine Did my best to interpret the corrections in https://github.com/filecoin-project/rust-fil-proofs/issues/827#issuecomment-528351096, but if this is still not the implementation required I'm going to suggest writing the specification _first_, and then I can implement whatever lands there.

**Comment by schomatis** [comment]
Will be included in https://github.com/filecoin-project/rust-fil-proofs/pull/864.

**Comment by dignifiedquire** [comment]
@schomatis can you please change this, such that the logic is all in expanded_parents/generate_expanded_parents, for the zigzag commitments work I need the parents() call to be only concatenating the base and expander parents

**Comment by schomatis** [comment]
Closing in favor of https://github.com/filecoin-project/rust-fil-proofs/pull/888 which will be directly merged into #864.

**Pull request closed without merging**